libutils/sequence.c

```diff
@@ -58,16 +59,19 @@ static void DestroyRange(Seq *seq, size_t start, size_t end)
 
 void SeqDestroy(Seq *seq)
 {
-    if (seq && seq->length > 0)
+    if (seq != NULL)
     {
-        DestroyRange(seq, 0, seq->length - 1);
+        if (seq->length > 0)
+        {
+            DestroyRange(seq, 0, seq->length - 1);
+        }
+        SeqSoftDestroy(seq);
     }
-    SeqSoftDestroy(seq);
 }
```

Comment:
Not a big deal, but I think this is more clear with an early return, instead of nested ifs.

**Original:**

```C
void SeqDestroy(Seq *seq)
{
    if (seq && seq->length > 0)
    {
        DestroyRange(seq, 0, seq->length - 1);
    }
    SeqSoftDestroy(seq);
}
```

**Your refactor:**


```C
void SeqDestroy(Seq *seq)
{
    if (seq != NULL)
    {
        if (seq->length > 0)
        {
            DestroyRange(seq, 0, seq->length - 1);
        }
        SeqSoftDestroy(seq);
    }
}
```

**My suggestion:**

```C
void SeqDestroy(Seq *seq)
{
    if (seq == NULL)
    {
        return;
    }

    if (seq->length > 0)
    {
        DestroyRange(seq, 0, seq->length - 1);
    }
    SeqSoftDestroy(seq);
}
```

Just an FYI.

Reply:
More interesting reading: https://refactoring.guru/replace-nested-conditional-with-guard-clauses

Reply:
Thanks for the advise. The early return recommendation is something I am aware of, I'll have it in mind next time